app/Jobs/IndexModel.php

```diff
@@ -42,9 +46,51 @@ final protected function execute(string $indexName): void
 
         // @phpstan-ignore-next-line
         $query->searchable();
+
+        if ($query->doesntExist()) {
+            return;
+        }
```

Comment:
is it the `searchable` method which indexes the data? If so, could the return check be moved above?